include/openssl/tls1.h

```diff
@@ -24,6 +24,9 @@ extern "C" {
 #  define OPENSSL_TLS_SECURITY_LEVEL 1
 # endif
 
+#ifndef OPENSSL_NO_CNSM
+# define SM1_1_VERSION                   0x0101
+#endif
```

Comment:
So this is a different protocol than SSL/TLS, looks a lot like TLS, but uses a different version number?
@richsalz or @kaduk, do you care to comment on this?

Reply:
This is not TLS.  It is a national variant. Does OpenSSL want to support this?
It uses many values that the IETF consider "its own" as defined in the registries at https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml  The handshake/version number here, but also the cipher assignments used elsewhere in the code.

My recommendation has two parts. First, someone (perhaps someone at TASS?) should apply to the IANA registry and get assigned numbers. Once that is done, the project should decide if it wants to implement this. 

As it stands now, if you implement this, you could run into interoperability problems if/when IETF TLS assigns conflicting curve, etc., values, and I recommend rejecting this PR.

Hope this info helps you decide.

Reply:
So one of the things about this PR is that they add a CNTLS_client_method(), but don't add a CNTLS_server_method(), and I think you just need to use the TLS_server_method(), and I think the TLS_server_method() they just accepts some other protocol.

I think they should either fully use a different protocol with different methods (like TLS vs DTLS), or actually use TLS and add the needed extensions to that.

Reply:
Yes, indeed the CNSM is really a national variant. It follow the main of tls1.1, but it has lots of different from the official tls. Such as the version num, use double certificate when handshake (When encryption is required, use the encryption certificate. When authentication is required, use the authentication certificate.)  We have the CNSM internally, so we develop the tassl for use by ourself. After we use, we also want to contribute the tassl to the official for those needed especially we see the  cryptographic algorithm of 'smx' series are released by the official openssl.
As for the follow-up, we are waiting for your decision and see our plan. Thank all of you! @kroeckx @richsalz @mspncp

Reply:
Since the IETF plans to deprecate TLS 1.0 and 1.1 (https://tools.ietf.org/html/draft-ietf-tls-oldversions-deprecate), and this at least seems to have some of the same issues, I don't think we should add it.

Reply:
After our discussion, we feel we should close this pull request. But we still want to thank you for helping us make some improvements and fix some bugs.